discovery-infra/update_assisted_service_cm.py

```diff
@@ -56,11 +75,29 @@ def _get_relevant_envs():
     return data
 
 
+def update_requirements(requirements_json):
+    try:
+        requirements = json.loads(requirements_json)
+        for version_requirements in requirements:
+            if version_requirements["version"] == "default":
+                version_requirements["master"] = MASTER_REQUIREMENTS
+                version_requirements["worker"] = WORKER_REQUIREMENTS
+    except json.JSONDecodeError:
+        requirements = REQUIREMENTS
```

Comment:
I would prefer an explicit failure (`raise`) if the JSON doesn't decode, rather than secretly switching it up.

Reply:
Do you know if the script is ran always against already pre-processed `assisted-service-configmap.yaml` file? If not, then the variable value that is saved in the repository won't be parsed (`HW_VALIDATOR_REQUIREMENTS: REPLACE_HW_VALIDATOR_REQUIREMENTS`).

Also - changes on the assisted-service side that process that the configmap definition file (https://github.com/openshift/assisted-service/pull/1458) are not merged yet so using defaults instead of failing provides backward compatibility.

Reply:
Not sure about the answer to your question, if this is too much trouble, then maybe a warning + a `# TODO: Consider raising here in the future to avoid implicitly ignoring user errors` would be enough for now.

If you know of specific, non-json values that this var might have then maybe add a special case for them (`if .. == known_value: use_defaults`)